sdcm/utils/net.py

```diff
@@ -31,8 +31,8 @@ def get_my_public_ip() -> str:
             result.raise_for_status()
 
             ip_address = result.text.strip()
-            ipaddress.ip_address(ip_address)  # validating that we got IP address
-        except (ValueError, requests.exceptions.RequestException):
+            ipaddress.IPv4Address(ip_address)  # validating that we got IPv4 address
+        except (ipaddress.AddressValueError, ipaddress.NetmaskValueError, requests.exceptions.RequestException):
             if idx == len(hostnames) - 1:
```

Comment:
Since we want to go over all services why then we do not consider the presence of an already detected IP address in the `ip_address` var?

For the moment the last `https://myip.dnsomatic.com` service is unresponsive and we will get unneeded exception on the line 37.

Reply:
anyhow that code was trying to make sure we propagate the actual failure back.
but was doing excess calls to all of those services.

since we use that in one single case, we didn't noticed too many issue with it.